robot.js

```diff
@@ -81,83 +81,129 @@
       const tr = document.createElement('tr');
       table.appendChild(tr);
       let rowHtml = '';
+      let cell = '';
       for (let col = 0; col < cells.length; col++) {
-        const cell = cells[col] === '.' ? '' : cells[col];
-        rowHtml += `<td>${cell}</td>`;
+        if ( cells[col] === '.') {
+           cell = '';
+      } else {
+        cell = cells[col];
       }
-      tr.innerHTML = rowHtml;
+      //const cell = cells[col] === '.' ? '' : cells[col];
+      rowHtml += `<td>${cell}</td>`;
     }
+    tr.innerHTML = rowHtml;
   }
+}
 
   function move() {
-    console.log('executing move()');
-    let x = robot.x;
-    let y = robot.y;
-
-    switch (robot.dir) {
-      case 'up':
-        y = y < board.length - 1 ? y + 1 : y;
-        break;
-      case 'down':
-        y = y > 0 ? y - 1 : y;
-        break;
-      case 'left':
-        x = x > 0 ? x - 1 : x;
-        break;
-      case 'right':
-        x = x < board[0].length - 1 ? x + 1 : x;
-        break;
-    }
+  console.log('executing move()');
+  let x = robot.x;
+  let y = robot.y;
+
+  switch (robot.dir) {
+    case 'up':
+      if (y < board.length - 1) {
+        y = y + 1;
+      } else {
+        y;
+      }
+      //y = y < board.length - 1 ? y + 1 : y;
```

Comment:
In this case you can leave out the `else` block because `y` already has the correct value. In fact, the statement `y;` by itself does nothing at all (like talking to yourself with nobody listening :grin:). If you want to completely be equivalent to the ternary operator your could do:

```js
y = y;
```

But that would be silly, no?

Same for the ternary operators.